tpi/base.py

```diff
@@ -65,3 +75,22 @@ async def _shell_async(self, *args, **kwargs):
                 stdout=sys.stdout,
                 stderr=sys.stderr,
             )
+
+    def _shell_default(
+        *args, host=None, username=None, port=None, client_keys=None, **kwargs
+    ):
+        assert host
+
+        cmd = ["ssh"]
+        if client_keys:
+            if isinstance(client_keys, str):
+                client_keys = [client_keys]
+            cmd.extend(
+                itertools.chain.from_iterable(
+                    ["-i", keyfile] for keyfile in client_keys
+                )
+            )
```

Comment:
is this better since `funcy` is a dependency already?

```suggestion
            cmd.extend(funcy.join(["-i", keyfile] for keyfile in client_keys))
```

or

```suggestion
            cmd.extend(
                funcy.merge(*(["-i", keyfile] for keyfile in client_keys))
            )
```

Reply:
Simpler one:
```suggestion
            cmd.extend(lcat(zip(repeat("-i"), client_keys)))
```

Or, if we are trying to be more functional:
```suggestion
           cmd += lcat(zip(repeat("-i"), client_keys))
```